### PR TITLE
Fix the issue when redirect request is triggered only after connection inactivity timeout

### PR DESCRIPTION
**Pull request by soylent**
Solves issue #276.

I am not sure if my solution is good. It keeps the existing behaviour except that redirect request is triggered right away.

### CODE REVIEW

File: lib/em-http/client.rb [hunk 1]
```diff
@@ -103,6 +103,7 @@ def unbind(reason = nil)
               @req.set_uri(@response_header.location)
 
               @conn.redirect(self)
+              @conn.conn.close_connection
```
Comment (igrigorik) [inline lib/em-http/client.rb]: This breaks the same-origin redirect case.

Comment (soylent) [inline lib/em-http/client.rb]: Thank you for your feedback. Any suggestions how to resolve this issue?

Comment (soylent) [inline lib/em-http/client.rb]: @igrigorik could you please elaborate? A test case would be very helpful.

**Comment by igrigorik** [comment]
@soylent sorry about the delay, had to dig in a bit to sort this one out.. I think this should do it: https://github.com/igrigorik/em-http-request/commit/de39103c9c269fabb00e842b4f630657d19bd070

Can you test above branch for your use case and (hopefully :)) confirm that its working?

**Comment by soylent** [comment]
@igrigorik it resolves the issue, thank you!

**Comment by igrigorik** [comment]
Excellent, should be in master now!

**Pull request merged**